runtime/compare.c

```diff
@@ -116,11 +142,13 @@ static intnat compare_val(value v1, value v2, int total)
 static intnat do_compare_val(struct compare_stack* stk,
                              value v1, value v2, int total)
 {
-  struct compare_item * sp;
+  struct compare_item * next_stack_slot;
```

Comment:
Renaming `sp` into `next_stack_slot` doesn't help.  A stack pointer is ALWAYS named `sp`, and if the stack grows upward, which is the case here, it ALWAYS points just above the latest stack entry, so that pushing `v` is `*sp++ = v` and popping `v` is `v = *--sp`.  These conventions are ingrained in my mind and in that of many systems programmers, so please follow them instead of inventing your own names.

Reply:
This is embarrassing, even though I did not write the old code: yes, the old code does not follow the `sp` convention I have in mind...  I agree with your fix, but would prefer `next_stack_slot` to be named `sp` and `sp` (in the line `struct compare_item* sp = next_stack_slot-1;`) to be named `item` or something like that.  Thanks for your patience.

Reply:
I'm not familiar with the convention regarding stacks and the `sp` variable, do we have that documented somewhere?
Having the code be (semi-)self-documenting is very helpful for those joining the codebase later, and helps with conveying the intentions in review. Naming the variable according to its intended usage (`next_stack_slot`) is a very easy way to accomplish that.

So maybe we can keep `next_stack_slot`, and call the item `item`? I agree that `item` is a much better name than `sp` in the context of the `compare_item` line

Reply:
I am planning to rename `next_stack_slot` back to `sp` to follow the implicit compiler-backend convention, and name the item `item`. Let's not waste our time discussing the details here when a code owner expresses a firm opinion.

Reply:
I renamed `next_stack_slot` back to `sp`. I use `next` as the item name when we are pushing a new item, and `last` as the item name when we are accessing the top of the stack.